src/facebook_client.py

```diff
@@ -0,0 +1,128 @@
+import json
+
+from core_data_modules.data_models import validators
+from core_data_modules.logging import Logger
+import requests
+from core_data_modules.traced_data import TracedData, Metadata
+from core_data_modules.util import TimeUtils
+from dateutil.parser import isoparse
+
+log = Logger(__name__)
+
+_BASE_URL = "https://graph.facebook.com/v8.0"
+_MAX_LIMIT = 100
+
+# TODO: Move to a new repo at AfricasVoices/SocialMediaTools
+# Included in project source for now because it's likely to evolve very rapidly during the initial experiments.
+class FacebookClient(object):
+    def __init__(self, access_token):
+        self._access_token = access_token
```

Comment:
To further harden the security here we can use the [appsecret_proof](https://developers.facebook.com/docs/graph-api/securing-requests) parameter and enable the setting to require proof on all calls. The app secret proof is a sha256 hash of our access token, using the app secret as the key. If you want to enable this now please let know so that I can switch on `Require proof on all calls` in the app settings.

Reply:
Moved to an offline discussion. Will add in a future PR if we decide to add this.